alacritty/src/event.rs

```diff
@@ -495,6 +497,56 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         }
     }
 
+    /// Update the cursor blinking state.
+    #[inline]
+    fn update_cursor_blinking(&mut self) {
+        // Get config cursor style.
+        let mut cursor_style = self.config.cursor.style;
+        if self.terminal.mode().contains(TermMode::VI) {
+            cursor_style = self.config.cursor.vi_mode_style.unwrap_or(cursor_style);
+        };
+
+        // Check terminal cursor style.
+        let terminal_blinking = self.terminal.cursor_style().blinking;
+        let blinking = cursor_style.blinking_override().unwrap_or(terminal_blinking);
+
+        // Update cursor blinking state.
+        self.scheduler.unschedule(TimerId::BlinkCursor);
+        if blinking && self.terminal.is_focused {
+            self.scheduler.schedule(
+                GlutinEvent::UserEvent(Event::BlinkCursor),
+                Duration::from_millis(self.config.cursor.blink_interval),
+                true,
+                TimerId::BlinkCursor,
+            )
+        } else {
+            *self.cursor_hidden = false;
+            self.terminal.dirty = true;
+        }
+    }
+
+    /// Enable input mode.
+    ///
+    /// Putting the user in input mode will temporarily disable some features like terminal cursor
+    /// blinking or the mouse cursor.
+    ///
+    /// This mode will disable itself automatically.
+    #[inline]
+    fn enable_input_mode(&mut self) {
```

Comment:
I'm not sure if I like `mode` here, maybe we should use something like `fn started_typing(&mut self)`. Since each time I see input mode I think it's IME stuff.

Reply:
I feel like a `started_typing` method should return a `bool` or something? I don't like that name since it doesn't really help me understand that it does.